### PR TITLE
Add v3.3 dockerfiles

### PR DESCRIPTION
**Pull request by ksylvest**
# Description
Support for [Ruby 3.3.0](https://www.ruby-lang.org/en/news/2023/12/25/ruby-3-3-0-released/).

# Reasons
To support the latest release of Ruby on CircleCI without needing to manually install.

# Checklist

Please check through the following before opening your PR. Thank you!

- [x] I have made changes to the `Dockerfile.template` file only
- [x] I have not made any manual changes to automatically generated files
- [x] My PR follows best practices as described in the [contributing guidelines](CONTRIBUTING)
- [x] (Optional, but recommended) My commits are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)

### CODE REVIEW

**Comment by schinery** [comment]
@ksylvest did you use `./shared/release.sh 3.3.0` to generate the files for this PR? 

Only asking as the PR title needs to include `[release]` in it for when it is merged and that script normally generates the PR title like "Publish v3.3.0. [release]". Details [here](https://github.com/CircleCI-Public/cimg-ruby?tab=readme-ov-file#building-the-dockerfiles).

File: 3.3/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,97 @@
+# vim:set ft=dockerfile:
```
Comment (willfish) [inline 3.3/Dockerfile:1]: I don't think you need this when the filename is `Dockerfile`

**Comment by ryanbourdais** [comment]
Closing in favor of https://github.com/CircleCI-Public/cimg-ruby/pull/153

**Comment by ksylvest** [comment]
@schinery I've adjusted title, but not sure if it matters given closed in favour of 153.

**Comment by JalexChen** [comment]
@ksylvest - some context:

@schinery's point about using the `./shared/release` script is correct. this is the section of the README that explains [it](https://github.com/CircleCI-Public/cimg-ruby?tab=readme-ov-file#publishing-official-images-for-maintainers-only). in doing so, we can make changes to the dockerfile without inadvertently releasing a new version.

We typically would use this PR, but Ryan provided an explanation in the other PR. We do appreciate the effort and I hope this doesn't discourage contributions from you in the future.

**Pull request closed without merging**